js/katex-a11y.js

```diff
@@ -0,0 +1,464 @@
+
+    genfrac: function(tree, a11yStrings) {
+        buildRegion(a11yStrings, function(a11yStrings) {
+            // NOTE: Not sure if this is a safe assumption
+            // hasBarLine true -> fraction, false -> binomial
+            if (tree.value.hasBarLine) {
+                a11yStrings.push("start fraction");
+                buildString(tree.value.leftDelim, "open", a11yStrings);
+                buildA11yStrings(tree.value.numer, a11yStrings);
+                a11yStrings.push("divided by");
+                buildA11yStrings(tree.value.denom, a11yStrings);
+                buildString(tree.value.rightDelim, "close", a11yStrings);
+                a11yStrings.push("end fraction");
+            } else {
+                a11yStrings.push("start binomial");
+                buildString(tree.value.leftDelim, "open", a11yStrings);
+                buildA11yStrings(tree.value.numer, a11yStrings);
+                a11yStrings.push("over");
+                buildA11yStrings(tree.value.denom, a11yStrings);
+                buildString(tree.value.rightDelim, "close", a11yStrings);
+                a11yStrings.push("end binomial");
+            }
+        });
+    },
+
+    // inner
+
+    katex: function(tree, a11yStrings) {
+        a11yStrings.push("KaTeX");
+    },
+
+    leftright: function(tree, a11yStrings) {
+        buildRegion(a11yStrings, function(a11yStrings) {
+            buildString(tree.value.left, "open", a11yStrings);
+            buildA11yStrings(tree.value.body, a11yStrings);
+            buildString(tree.value.right, "close", a11yStrings);
+        });
+    },
+
+    llap: function(tree, a11yStrings) {
+        buildA11yStrings(tree.value.body, a11yStrings);
+    },
+
+    mathord: function(tree, a11yStrings) {
+        buildA11yStrings(tree.value, a11yStrings);
+    },
+
+    op: function(tree, a11yStrings) {
+        buildString(tree.value.body, "normal", a11yStrings);
+    },
+
+    open: function(tree, a11yStrings) {
+        buildString(tree.value, "open", a11yStrings);
+    },
+
+    ordgroup: function(tree, a11yStrings) {
+        buildA11yStrings(tree.value, a11yStrings);
+    },
+
+    overline: function(tree, a11yStrings) {
+        buildRegion(a11yStrings, function(a11yStrings) {
+            a11yStrings.push("start overline");
+            buildA11yStrings(tree.value.body, a11yStrings);
+            a11yStrings.push("end overline");
+        });
+    },
+
+    phantom: function(tree, a11yStrings) {
+        a11yStrings.push("empty space");
+    },
+
+    punct: function(tree, a11yStrings) {
+        buildString(tree.value, "punct", a11yStrings);
+    },
+
+    rel: function(tree, a11yStrings) {
+        buildString(tree.value, "rel", a11yStrings);
+    },
+
+    rlap: function(tree, a11yStrings) {
+        buildA11yStrings(tree.value.body, a11yStrings);
+    },
+
+    rule: function(tree, a11yStrings) {
+        // NOTE: Is there something more useful that we can put here?
+        a11yStrings.push("rule");
```

Comment:
Hmm - maybe! As far as I can tell the height and width of this is variable, so it's not always box-like. Considering that it's not used (at least not in our sample set) it probably doesn't matter what we name it. I think I'll leave it as "rule" for now and we can re-explore this later.

https://en.wikibooks.org/wiki/LaTeX/Rules_and_Struts